Adds experimental features and OCI storage (#179)
rancher/fleet-docs: docs/enableexperimental.md

# How to enable experimental features

Fleet supports experimental features that are disabled by default and that can be enabled by the user.

> Enabling/disabling experimental feaures is done usign extra environment variales that are available when deploying rancher/fleet.

Reviewer: ```suggestion
Enabling/disabling experimental features is done using extra environment variables that are available when deploying `rancher/fleet`.
```

Apart from typo fixes, this adds backticks around `rancher/fleet` to clarify that we're talking about a chart. Unless the intended meaning was rather something like "Rancher or Fleet", as environment variables would be available in both cases...?

## Enabling an experimental feature

At the moment we're writing this document fleet has OCI storage as an experimental feature.

> ### Enabling when installing fleet stand-alone

Reviewer: nit: an extra line break here makes the general layout of this document look nicer to me, but this comment can be safely dismissed :)
```suggestion

### Enabling when installing fleet stand-alone
```

All you need to do is to pass something like:
```
--set-string extraEnv[0].name=EXPERIMENTAL_OCI_STORAGE \
--set-string extraEnv[0].value=true \
```
to your helm install or update command. 

Please note you have to use `--set-string` because otherwise the boolean value won't work as expected.

> ### Enabling when installing fleet with rancher
> 
> You can also activate the experimental features in `Fleet` when installing `Rancher`.

Reviewer: I think it would be interesting to discuss the use of upper- vs lowercase first letter in naming, ie `[Ff]leet` and `[Rr]ancher`, as I'd do the opposite here:
(rationale: using uppercase when referring to products/programs, but lowercase when referring to chart names as they'd appear on the command line; not sure if that's shared practice though, hence the interest in a (brief) discussion)
```suggestion
### Enabling when installing Fleet with Rancher

You can also activate the experimental features in `fleet` when installing `rancher`.
```

# How to enable experimental features

Fleet supports experimental features that are disabled by default and that can be enabled by the user.

Enabling/disabling experimental features is done using extra environment variables that are available when deploying `rancher/fleet`.

## Enabling an experimental feature

> At the moment we're writing this document `Fleet` has OCI storage as an experimental feature.

Reviewer: ```suggestion
At the moment we're writing this document, Fleet has OCI storage as an experimental feature.
```
Why the backticks around `Fleet`? "Fleet" with an uppercase "F" is not the name of a chart nor a repo, it wouldn't appear as such on the command line.

### Enabling when installing Fleet stand-alone

All you need to do is to pass something like:
```
--set-string extraEnv[0].name=EXPERIMENTAL_OCI_STORAGE \
--set-string extraEnv[0].value=true \
```
to your helm install or update command. 

Please note you have to use `--set-string` because otherwise the boolean value won't work as expected.

### Enabling when installing Fleet with Rancher

> You can also activate the experimental features in `Fleet` when installing `Rancher`.

Reviewer: Same remark/question as above:
```suggestion
You can also activate the experimental features in `fleet` when installing `rancher`.
```
or:

```suggestion
You can also activate the experimental features in Fleet when installing Rancher.
```

> The parameters are the same, but you have to add the `Fleet.` prefix.

Reviewer: Would an uppercase "F" even be correct here? This looks like an accidental search-and-replace hit.
```suggestion
The parameters are the same, but you have to add the `fleet.` prefix.
```

```
--set-string fleet.extraEnv[0].name=EXPERIMENTAL_OCI_STORAGE \
--set-string fleet.extraEnv[0].value=true \
```

## Available experimental features

> Right now `Fleet` supports the following experimental features:

Reviewer: ```suggestion
Right now Fleet supports the following experimental features:
```
